### PR TITLE
Bug 1907621: Bump cluster-api-provider-kubevirt version

### PR DESCRIPTION
**Pull request by nirarg**
Changed `cluster-api-provider-kubevirt` to include newer version of `machine-api-operator` module

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -115,7 +115,7 @@ require (
 	sigs.k8s.io/cluster-api-provider-aws v0.0.0
 	sigs.k8s.io/cluster-api-provider-azure v0.0.0
 	sigs.k8s.io/cluster-api-provider-openstack v0.0.0
-	sigs.k8s.io/controller-tools v0.3.1-0.20200617211605-651903477185
+	sigs.k8s.io/controller-tools v0.4.1
```
Comment (staebler) [inline go.mod:118]: We cannot update controller-tools to v0.4.1, without figuring out what to do about the `IPNet` struct. You can see why if you try to run `./hack/verify-codegen.sh`. You will need to add a replace for this.

**Review by staebler — COMMENTED** [review]
Please remove the `./vendor2` directory as well. That was created to carry a local version of cluster-api-provider-kubevirt.

**Review by staebler — COMMENTED** [review]
/approve
/lgtm

**Comment by staebler** [comment]
Need to run `go generate ./pkg/types/installconfig.go`

/lgtm cancel

**Comment by nirarg** [comment]
Hi @staebler the `go generate` changes the annotation `controller-gen.kubebuilder.io/version` from `v0.3.1-0.20200617211605-651903477185` to `v0.4.1` 
Is it expected?

**Comment by staebler** [comment]
> Hi @staebler the `go generate` changes the annotation `controller-gen.kubebuilder.io/version` from `v0.3.1-0.20200617211605-651903477185` to `v0.4.1`
> Is it expected?

Unfortunately, yes. The version is coming from the go mod info and is not respecting the replace stanza.

**Pull request merged**